### PR TITLE
Docs: Setup directories with correct permissions

### PR DESCRIPTION
**Pull request by vagishagupta23**
Documentation updated with the steps to setup the directories with
the correct permissions.

Make sure these boxes are signed before submitting your Pull Request
-- thank you.

- [x] I have read the contributing guide lines at
  https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation
  contribution agreement at
  https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the
  changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/3038

### CODE REVIEW

File: doc/quickstart.rst [hunk 1]
```diff
@@ -40,20 +40,55 @@ Directories and Permissions
...
+    sudo usermod -a -G suricata username
+
+Verify whether group has been changed::
+
+    ls -al /etc/suricata
+    ls -al /var/lib/suricata/rules
+    ls -al /var/lib/suricata/update
+
+Reboot your system. Run ``suricata-update`` without a sudo to check
+if suricata-update functions.
```
Comment (shivan1b) [inline doc/quickstart.rst]: @jasonish could you please have a look here?

**Comment by Kolano** [comment]
> Do not use the path ``/etc/suricata`` and ``/var/lib`` for example, since it could be any other path that user defines during configuration and is by default ``/usr/local``. Please check your configuration directory for the path and use that accordingly.

The wording of that isn't very good. I'd suggest...
> The paths ``/etc/suricata`` and ``/var/lib`` above are used in the default configuration and are dependent on paths set during compilation / the suricata.yaml and update.yaml files, By default these paths are set to ``/usr/local``. Please check your configuration for appropriate paths.

**Comment by shivan1b** [comment]
@Kolano thanks. Do you have something in mind that would seem more legible to you as a user?

**Comment by Kolano** [comment]
> @Kolano thanks. Do you have something in mind that would seem more legible to you as a user?

See above.

**Comment by shivan1b** [comment]
> > @Kolano thanks. Do you have something in mind that would seem more legible to you as a user?
> 
> See above.

Thanks. Sorry, I thought that was also a part of the commit. @vagishagupta23 could you please have a look at this?
I think what @Kolano has suggested makes more sense and is more legible.

**Comment by shivan1b** [comment]
@vagishagupta23 ping. Could you please make the doc addition more legible?
Thanks.

File: doc/quickstart.rst [hunk 2]
```diff
@@ -40,20 +40,55 @@ Directories and Permissions
...
+
+First, create a group ``suricata``::
+
+    sudo groupadd suricata
+
+Next, change the group of the directories and its files recursively::
+
+    sudo chgrp -R suricata /etc/suricata
+    sudo chgrp -R suricata /var/lib/suricata/rules
+    sudo chgrp -R suricata /var/lib/suricata/update
```
[the remark this answers is not in the document]
Comment (jasonish) [reply doc/quickstart.rst]: Shouldn't need to do this. Root would be preferable as the user, as we want this group to be what matters here. So I think it should be OK as is.

File: doc/quickstart.rst [hunk 3]
```diff
@@ -40,20 +40,55 @@ Directories and Permissions
...
+.. note:: Do not use the path ``/etc/suricata`` and ``/var/lib`` for
+	        example, since it could be any other path that user defines
+	        during configuration and is by default ``/usr/local``. Please
+	        check your configuration directory for the path and use that
+	        accordingly.
+
+Setup the directories with the correct permissions for the ``suricata``
+group::
+
+    sudo chmod -R g+rw /etc/suricata/
```
Comment (jasonish) [inline doc/quickstart.rst]: The group only needs read-only access to /etc/suricata.  So:
```
sudo chmod -R g+r /etc/suricata
```
should be enough.

**Comment by inashivb** [comment]
Rebased, issues fixed and commit added in #217 . Thanks, Vagisha!

**Pull request closed without merging**